### ISSUE TITLE
magit-direct-use-buffer-arguments 'always' is broken

### ISSUE THREAD

**Issue by nicketson** [magit/magit#4046]
According to the docs, setting `magit-direct-use-buffer-arguments` to `always` should re-use the current `magit-buffer-diff-args` and `magit-buffer-diff-files` of any existing status buffer when I call `magit-status`, regardless of whether it is visible or active.

Instead, any file filter I have for a status buffer is wiped out in between calls to `magit-status` if the status buffer is not visible when `magit-status` is called again.

I've tracked this down to the call to `magit-get-mode-buffer` in `magit-diff--get-value`. When `magit-direct-use-buffer-arguments` is `always`, the `FRAME` argument passed to `magit-get-mode-buffer` is `all`. Shouldn't it be `nil`?

My use case for this is that I have added a transient suffix command to `magit-diff-refresh` that sets `magit-buffer-diff-files` to exclude certain large build files (which take way too long to display) from the diffs in the status buffer. Prior to transient I had something similar that worked great but now I don't see any other way of doing this.

**Comment by tarsius** [comment]
Thanks for this nice report! I've applied the suggested fix and fixed some minor things I found along the way.

> According to the docs, setting `magit-direct-use-buffer-arguments` to `always` should re-use the current `magit-buffer-diff-args` and `magit-buffer-diff-files` of any existing status buffer when I call `magit-status`, regardless of whether it is visible or active.

I am considering treating the status buffer differently so that it always behaves like this, regardless of the value of this option. I think for diff. log, and derived buffers it can be confusing if you come back and the arguments you wanted to use only temporarilyare automatically resurrected hours later, while you might reasonably want that to nearly always happen for the status buffer. @kyleam Whatcha thinky?
